Test: Simplified Kafka K8S test

eloycoto opened the pull request:
- Add a livenessProbe in kafka manifest, wait until kafka is ready.
- Make test code clear
- Add CepGet in kubectl.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

manalibhutiyani commented on test/k8sT/KafkaPolicies.go:
```diff
@@ -107,157 +106,112 @@ var _ = Describe("K8sValidatedKafkaPolicyTest", func() {
-
-		By("Waiting for all Cilium Pods and endpoints to be ready ")
-		By("Waiting for node K8s1")
-		ciliumPod1, _ := kubectl.WaitCiliumEndpointReady(podFilter, helpers.K8s1)
```
@eloy: This is a multimode setup, with 

 ```
Kafka multinode setup:
 
 			K8s1:
 				1. empire-backup
 				2. empire-hq
 				3. kafka-broker : ingress policy only
 
 			K8s2:
 			   1. zook
 			   2. empire-outpost-8888
 			   3. empire-outpost-9999
```
 	

You are only testing for k8s2 nodes and have removed the logic for making sure all pods are up on k8s1

manalibhutiyani commented on test/k8sT/KafkaPolicies.go:
```diff
@@ -107,157 +106,112 @@ var _ = Describe("K8sValidatedKafkaPolicyTest", func() {
-		endpoints1, err := kubectl.CiliumEndpointsListByLabel(ciliumPod1, podFilter)
-		policyStatus1 := endpoints1.GetPolicyStatus()
-
-		/*
```
Why was this comment removed? This helps to understand the Kafka multimode setup. Could you please put it back?

manalibhutiyani commented on test/k8sT/KafkaPolicies.go:
```diff
@@ -107,157 +106,112 @@ var _ = Describe("K8sValidatedKafkaPolicyTest", func() {
-		By("Testing Kafka endpoint policy enforcement status on K8s1")
-		Expect(policyStatus1[models.EndpointPolicyEnabledNone]).Should(Equal(2))
-		// Only the kafka broker app should have ingress policy enabled.
-		Expect(policyStatus1[models.EndpointPolicyEnabledIngress]).Should(Equal(1))
```
Why are we not testing the `EndpointPolicyEnabled` status anymore?

manalibhutiyani commented on test/k8sT/KafkaPolicies.go:
```diff
@@ -45,24 +41,23 @@ var _ = Describe("K8sValidatedKafkaPolicyTest", func() {
-		backupApp = "empire-backup"
-		empireHqApp = "empire-hq"
-		outpostApp = "empire-outpost"
+		ciliumPod, err = kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s2)
```
Should we not be doing this for k8s1 ? This is a multinode setup?

manalibhutiyani reviewed (changes requested):
I think the fact that this is a multi-node setup is not taken into account. Please correct me if I am wrong. Also, the comment explaining the multinode setup helps people to understand the code better when reading it for the first time. Could you put that back.

manalibhutiyani commented on test/k8sT/manifests/kafka-sw-app.yaml:
```diff
@@ -28,6 +28,17 @@ spec:
+            port: 9092
+          initialDelaySeconds: 5
+          periodSeconds: 5
       nodeSelector:
```
I think we want to truly randomize it we need to remove the constaints on `nodeSelector:` of all the app pods

eloycoto committed:
Test: Simplified Kafka K8S test

- Add a livenessProbe in kafka manifest, wait until kafka is ready.
- Make test code clear
- Add CepGet in kubectl.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

manalibhutiyani commented on test/helpers/kubectl.go:
```diff
@@ -167,6 +167,29 @@ func CreateKubectl(vmName string, log *logrus.Entry) *Kubectl {
+	cmd := fmt.Sprintf("%s -n %s get cep %s -o json | jq '.status'", KubectlCmd, namespace, pod)
+	res := kub.Exec(cmd)
+	if !res.WasSuccessful() {
+		log.Debug("cep is not presnet")
```
nit `presnet` -> `present`

manalibhutiyani commented on test/k8sT/KafkaPolicies.go:
```diff
@@ -115,116 +97,86 @@ var _ = Describe("K8sValidatedKafkaPolicyTest", func() {
+
+		err := kubectl.ExecKafkaPodCmd(
+			helpers.DefaultNamespace, appPods[empireHqApp], prodHqAnnounce)
+		Expect(err).Should(BeNil(), "EmpireHQApp cannot announce HQ")
```
`EmpireHQApp cannot announce HQ` does not convey what Kafka message failed.
Please change this to 
`Failed to produce to empire-hq on topic empire-announce`

manalibhutiyani commented on test/k8sT/KafkaPolicies.go:
```diff
@@ -115,116 +97,86 @@ var _ = Describe("K8sValidatedKafkaPolicyTest", func() {
-			helpers.DefaultNamespace, appPods[outpostApp], fmt.Sprintf(conOutpostAnnoune))
-		Expect(err).Should(BeNil(), "Failed to consume from outpost on topic empire-announce")
+			helpers.DefaultNamespace, appPods[outpostApp], conOutpostAnnoune)
+		Expect(err).Should(BeNil(), "OutpostApp cannot announce Outpost")
```
Ditto. this does not convey what Kafka message and topic failed on which action. Please change the message back to `Failed to consume from outpost on topic empire-announce`

manalibhutiyani commented on test/k8sT/KafkaPolicies.go:
```diff
@@ -115,116 +97,86 @@ var _ = Describe("K8sValidatedKafkaPolicyTest", func() {
-			helpers.DefaultNamespace, appPods[empireHqApp], fmt.Sprintf(prodHqDeathStar))
-		Expect(err).Should(BeNil(), "Failed to produce to empire-hq on topic deathstar-plans")
+			helpers.DefaultNamespace, appPods[empireHqApp], prodHqDeathStar)
+		Expect(err).Should(BeNil(), "EmpireHqApp cannot produce messages")
```
Ditto:
`EmpireHqApp cannot produce messages` does not convey on which topic. Please change back to
`Failed to produce to empire-hq on topic deathstar-plans`

manalibhutiyani commented on test/k8sT/KafkaPolicies.go:
```diff
@@ -115,116 +97,86 @@ var _ = Describe("K8sValidatedKafkaPolicyTest", func() {
-			helpers.DefaultNamespace, appPods[outpostApp], fmt.Sprintf(conOutDeathStar))
-		Expect(err).Should(BeNil(), "Failed to consume from outpost on topic deathstar-plans")
+			helpers.DefaultNamespace, appPods[outpostApp], conOutDeathStar)
+		Expect(err).Should(BeNil(), "OutpostApp cannot produce messages")
```
Ditto. `OutpostApp cannot produce messages` does not convey the topic. Please change back to `Failed to consume from outpost on topic deathstar-plans`

manalibhutiyani commented on test/k8sT/KafkaPolicies.go:
```diff
@@ -115,116 +97,86 @@ var _ = Describe("K8sValidatedKafkaPolicyTest", func() {
-			helpers.DefaultNamespace, appPods[backupApp], fmt.Sprintf(prodBackAnnounce))
-		Expect(err).Should(BeNil(), "Failed to produce to backup on topic empire-announce")
+			helpers.DefaultNamespace, appPods[backupApp], prodBackAnnounce)
+		Expect(err).Should(BeNil(), "BackupApp cannot not announce")
```
`BackupApp cannot not announce` ---> `Failed to produce to backup on topic empire-announce`
This specifies exactly what action on what topic failed for the particular apps.

manalibhutiyani commented on test/k8sT/KafkaPolicies.go:
```diff
@@ -115,116 +97,86 @@ var _ = Describe("K8sValidatedKafkaPolicyTest", func() {
-
-		trace := kubectl.CiliumExec(ciliumPod1, fmt.Sprintf(
+			helpers.DefaultNamespace, appPods[outpostApp], prodOutAnnounce)
+		Expect(err).Should(BeNil(), "OutpostApp cannot not annouce")
```
`OutpostApp cannot not annouce` does not make a lot of sense to me.
Maybe change to `Failed to produce to outpost on topic empire-announce`
